### PR TITLE
Docs: Update `DIRECTORY_STRUCTURE.md`

### PR DESCRIPTION
**Pull request by nick-y-ito**
## Overview

This PR syncs the current directory structure to `DIRECTORY_STRUCTURE.md`.

## Changes

- Synced the current directory structure to `DIRECTORY_STRUCTURE.md`
- Deleted unnecessary directories that were kept with `.gitkeep`

## Review points

- Is there any wrong description?
- Is there any missed directory?

## Notes

-

## Assignee Checklist:

- [x] The naming convention of the PR title is correct (See the comment at the top of this template)
- [x] The base branch is correct (See: [Types of Branches](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#types-of-branches))
- [x] The branch name follows the [Branch Naming Conventions](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#branch-naming-conventions)
- [x] The correct assignees and reviewers have been designated for this PR
- [x] The coding style follows the [Coding Style Guide](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [x] All the related issues are associated with this PR
- [x] All criteria in the associated issues are met (please tick the checkboxes)
- [x] My changes do not generate new warnings or errors (especially in the console)

## Reviewer Checklist:

- [ ] Double-check the "Assignee Checklist"
- [ ] The code follows the generic best practices and our [Coding Style Guide](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [ ] The code is well-commented and easy to understand
- [ ] The UI changes accurately reflect the provided design

### CODE REVIEW

File: docs/DIRECTORY_STRUCTURE.md [hunk 1]
```diff
@@ -1,24 +1,27 @@
...
+  - [Overview](#overview)
     - [`app/`](#app)
+    - [`assets/`](#assets)
+    - [`assets/images/`](#assetsimages)
+    - [`assets/images/icons/`](#assetsimagesicons)
+    - [`assets/images/logos/`](#assetsimageslogos)
     - [`components/`](#components)
     - [`components/layouts/`](#componentslayouts)
-    - [`components/icons/`](#componentsicons)
     - [`components/pages/`](#componentspages)
```
Comment (kanta1207) [inline docs/DIRECTORY_STRUCTURE.md:16]: Minor point but it's `component/page` in actual directory structure.

Should we change the directory name at some point to meet with this document?

Comment (nick-y-ito) [reply docs/DIRECTORY_STRUCTURE.md:16]: Oh, I didn't notice the mismatch!

I think `components/pages/` is more natural. Sounds good?

**Review by kanta1207 — APPROVED** [review]
Thank you for updating! 
The document itself is looking perfect and good to go, but one thing I noticed is that we have to rename `components/page` to `components/pages` at some point!
Mentioned in #313

**Comment by nick-y-ito** [comment]
@kanta1207 
Thank you for your review and reporting the mismatch with the actual directory name.

I think it can be fixed in this PR because it matters to this issue and it won't cause so complex conflict. How does it sound?

File: docs/DIRECTORY_STRUCTURE.md [hunk 2]
```diff
@@ -1,24 +1,27 @@
...
 Welcome to our project! This document outlines our project's directory structure, adhering to a "Feature-Driven Folder Structure" approach. The structure is inspired by insights from both an [English article](https://dev.to/profydev/screaming-architecture-evolution-of-a-react-folder-structure-4g25#indexjs-as-public-api) and a [Japanese article](https://zenn.dev/necscat/articles/d5d9b7a3f859d7).
 
-## Directory Structure Overview
+## Overview
 
-- [Project Directory Structure](#project-directory-structure)
-  - [Directory Structure Overview](#directory-structure-overview)
+- [Directory Structure](#directory-structure)
+  - [Overview](#overview)
     - [`app/`](#app)
```
Comment (ShoeheyOt) [inline docs/DIRECTORY_STRUCTURE.md:9]: Is it not needed to mention inside of `app/` directory ?

File: docs/DIRECTORY_STRUCTURE.md [hunk 3]
```diff
@@ -1,24 +1,27 @@
...
+    - [`assets/`](#assets)
+    - [`assets/images/`](#assetsimages)
+    - [`assets/images/icons/`](#assetsimagesicons)
+    - [`assets/images/logos/`](#assetsimageslogos)
```
Comment (ShoeheyOt) [inline docs/DIRECTORY_STRUCTURE.md:13]: What about `ui` directory inside `assets/images`?

File: docs/DIRECTORY_STRUCTURE.md [hunk 4]
```diff
@@ -33,78 +36,92 @@ The following is a brief overview of the project's directory structure under the
...
 
-Contains non-shared utility functions relevant to the feature.
+Stores feature-specific utility functions.
 
 ### `hooks/`
 
-Shared custom hooks utilized across various features.
+Stores all [custom hooks](https://react.dev/learn/reusing-logic-with-custom-hooks) used across the project.
 
 ### `lib/`
```
Comment (ShoeheyOt) [inline docs/DIRECTORY_STRUCTURE.md:113]: Is it not needed to mention we store API calling file  in `lib` directory?

**Review by ShoeheyOt — COMMENTED** [review]
@nick-y-ito Thank you for updating the file. I have three points which is curious to me.
I would appreciate hearing from you :)
Thanks!

File: docs/DIRECTORY_STRUCTURE.md (hunk 2 above)
Comment (nick-y-ito) [reply docs/DIRECTORY_STRUCTURE.md:9]: This directory is for routing, and all the other sub-directories are corresponded to the URL paths. Therefore, they are really not suitable for writing here.

File: docs/DIRECTORY_STRUCTURE.md (hunk 3 above)
Comment (nick-y-ito) [reply docs/DIRECTORY_STRUCTURE.md:13]: Oh, I was referring to the older `develop` branch, which didn't have `assets/images/ui/`. I will add it!

File: docs/DIRECTORY_STRUCTURE.md (hunk 4 above)
Comment (nick-y-ito) [reply docs/DIRECTORY_STRUCTURE.md:113]: Oh, absolutely! That was very important.

**Comment by nick-y-ito** [comment]
@kanta1207 @ShoeheyOt 
I have added descriptions for these directories:

- `assets/images/logos/`
- `lib/api/`
- `lib/api/<resource_name>/`
- `lib/api/<resource_name>/client/`
- `lib/api/<resource_name>/server/`

Could you please review it again, please?

**Review by ShoeheyOt — APPROVED** [review]
@nick-y-ito Thank you for quick actions! LGTM

**Comment by kanta1207** [comment]
> I think it can be fixed in this PR because it matters to this issue and it won't cause so complex conflict. How does it sound?

Thank you for sorting out and sorry for the late reply. 
As you're saying, it's also totally fine to just rename the directory name to match with the document.

Can you resolve with the issue in this PR, please 🙏

**Comment by nick-y-ito** [comment]
@kanta1207 @ShoeheyOt 
Thank you for your reviews :)

**Pull request merged**